packages/vm/src/imports.rs

```diff
@@ -235,6 +258,34 @@ fn do_humanize_address<A: Api, S: Storage, Q: Querier>(
     }
 }
 
+fn do_verify_secp256k1<A: Api, S: Storage, Q: Querier>(
+    env: &Environment<A, S, Q>,
+    hash_ptr: u32,
+    signature_ptr: u32,
+    pubkey_ptr: u32,
+) -> VmResult<u32> {
+    let hash = Binary(read_region(&env.memory(), hash_ptr, LENGTH_SHA256_HASH)?);
+
+    let signature = Binary(read_region(
+        &env.memory(),
+        signature_ptr,
+        MAX_LENGTH_SIGNATURE,
+    )?);
+
+    let pubkey = Binary(read_region(&env.memory(), pubkey_ptr, MAX_LENGTH_PUBKEY)?);
```

Comment:
Why do you wrap those as `Binary`? We can simply use the `Vec<u8>` from `read_region`

Reply:
Copied from human/canonical addr I guess. I'll simplify it.